edgexfoundry/device-virtual-go
# refactor(snap): edgex-snap-hooks related upgrade

This PR upgrades snap hooks to [2.3.0 ](https://github.com/canonical/edgex-snap-hooks/releases/tag/v2.3.0)with new function and refactor:
- Remove export bash command to process valid .env files
- Use Fatal for logging
- Use library function for copying directory

Signed-off-by: Mengyi Wang <mengyi.wang@canonical.com>

TODO:
- [x] switch to `github.com/canonical/edgex-snap-hooks/v2 v2.3.0` when available: https://github.com/canonical/edgex-snap-hooks/pull/59







**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-virtual-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  <link to docs PR>

## Testing Instructions

1. Install and setup:
```
sudo snap install edgex-device-virtual --channel=edge/pr-290
sudo snap set edgex-device-virtual config.writable-loglevel=DEBUG
sudo snap set edgex-device-virtual config.service-startupmsg="#test start up message#"
sudo snap start edgex-device-virtual
```
2. Check the .env file is valid now without `export` command:
```
cat /var/snap/edgex-device-virtual/current/config/device-virtual/res/device-virtual.env 

SERVICE_STARTUPMSG="#test start up message#"
WRITABLE_LOGLEVEL="DEBUG"
```
3. Validate env vars in `.env` have been exported successfully:
```
sudo snap logs -f edgex-device-virtual

level=INFO ts=2022-06-14T14:06:02.834014222Z app=device-virtual source=message.go:55 msg="#test start up message#"
level=DEBUG ts=2022-06-14T14:06:12.834954223Z app=device-virtual source=executor.go:52 msg="AutoEvent - reading Bool"
```
4. Validate all config files have been copied in right place:
```
tree /var/snap/edgex-device-virtual/current/config
.
└── device-virtual
    └── res
        ├── configuration.toml
        ├── devices
        │   └── devices.toml
        ├── device-virtual.env
        └── profiles
            ├── device.virtual.binary.yaml
            ├── device.virtual.bool.yaml
            ├── device.virtual.float.yaml
            ├── device.virtual.int.yaml
            └── device.virtual.uint.yaml

4 directories, 8 files
```


## New Dependency Instructions (If applicable)
<!-- Please follow [vetting instructions](https://wiki.edgexfoundry.org/display/FA/Vetting+Process+for+3rd+Party+Dependencies) and place results here --
